### ISSUE TITLE
Invalid SSE Header doing multipart upload to S3 with TransferManager

### ISSUE THREAD

**Issue by mstockm** [aws/aws-sdk-java#1876]
We are using JRE version 1.8.0_151. We upgraded from 1.11.470 to 1.11.475 and started getting the following error when using this method of TransferManager https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/s3/transfer/TransferManager.html#upload-com.amazonaws.services.s3.model.PutObjectRequest- :

```
com.amazonaws.SdkClientException: Unable to complete multi-part upload. Individual part upload failed : x-amz-server-side-encryption header is not supported for this operation. (Service: Amazon S3; Status Code: 400; Error Code: InvalidArgument; Request ID: 5506E77092B77852; S3 Extended Request ID: RZJsAOjdc9kLDeRaLEOYk8vlZsBaK6qtMr5OgC+N9ildaWXLQz9knR0W3K/KrDL1QPgjRbl++4g=)
	at com.amazonaws.services.s3.transfer.internal.CompleteMultipartUpload.collectPartETags(CompleteMultipartUpload.java:129)
	at com.amazonaws.services.s3.transfer.internal.CompleteMultipartUpload.call(CompleteMultipartUpload.java:90)
	at com.amazonaws.services.s3.transfer.internal.CompleteMultipartUpload.call(CompleteMultipartUpload.java:41)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	... 3 more
Caused by: com.amazonaws.services.s3.model.AmazonS3Exception: x-amz-server-side-encryption header is not supported for this operation. (Service: Amazon S3; Status Code: 400; Error Code: InvalidArgument; Request ID: 5506E77092B77852; S3 Extended Request ID: RZJsAOjdc9kLDeRaLEOYk8vlZsBaK6qtMr5OgC+N9ildaWXLQz9knR0W3K/KrDL1QPgjRbl++4g=), S3 Extended Request ID: RZJsAOjdc9kLDeRaLEOYk8vlZsBaK6qtMr5OgC+N9ildaWXLQz9knR0W3K/KrDL1QPgjRbl++4g=
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.handleErrorResponse(AmazonHttpClient.java:1695)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeOneRequest(AmazonHttpClient.java:1350)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeHelper(AmazonHttpClient.java:1101)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.doExecute(AmazonHttpClient.java:758)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeWithTimer(AmazonHttpClient.java:732)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.execute(AmazonHttpClient.java:714)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.access$500(AmazonHttpClient.java:674)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutionBuilderImpl.execute(AmazonHttpClient.java:656)
	at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:520)
	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:4705)
	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:4652)
	at com.amazonaws.services.s3.AmazonS3Client.doUploadPart(AmazonS3Client.java:3657)
	at com.amazonaws.services.s3.AmazonS3Client.uploadPart(AmazonS3Client.java:3642)
	at com.amazonaws.services.s3.transfer.internal.UploadPartCallable.call(UploadPartCallable.java:64)
	at com.amazonaws.services.s3.transfer.internal.UploadPartCallable.call(UploadPartCallable.java:31)
	... 4 more
```

This only happened for uploads above a certain size (presumably when the underlying request shifted to multipart?)

We are generating the PutObjectRequest as follows:

```java
        PutObjectRequest request = new PutObjectRequest(bucket, key, file);
        ObjectMetadata metadata = new ObjectMetadata();

        metadata.setSSEAlgorithm(ObjectMetadata.AES_256_SERVER_SIDE_ENCRYPTION);
        metadata.setContentEncoding(GZIP_ENCODING);
        metadata.setContentType(contentType);

        request.setMetadata(metadata);

        return request;
```

**Comment by debora-ito** [comment]
Hi @mstockm, we are aware of the issue and a fix was released in `1.11.481`, can you please check it and see if it works in your case?

**Comment by mstockm** [comment]
Yup I can confirm that we no longer see the behavior in `1.11.481` - thanks!

**Comment by cl305** [comment]
Hey all, sorry to reopen this, but I'm also running into a similar issue. I'm currently using SDK version `1.11.489` and am receiving the following error:

```
com.amazonaws.services.s3.model.AmazonS3Exception: x-amz-server-side-encryption header is not supported for this operation. (Service: Amazon S3; Status Code: 400; Error Code: InvalidArgument; Request ID: 2FD3C49FFCC4A36B; S3 Extended Request ID: p9lFO7r+VJECeqhTMdFJTXdN3OPAEgcELBMgz5EGF6Wh+SzhTK7mHdxEBMIsHKm8AIv7H8SXJW4=), S3 Extended Request ID: p9lFO7r+VJECeqhTMdFJTXdN3OPAEgcELBMgz5EGF6Wh+SzhTK7mHdxEBMIsHKm8AIv7H8SXJW4=
        at com.amazonaws.http.AmazonHttpClient$RequestExecutor.handleErrorResponse(AmazonHttpClient.java:1695)
        at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeOneRequest(AmazonHttpClient.java:1350)
        at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeHelper(AmazonHttpClient.java:1101)
        at com.amazonaws.http.AmazonHttpClient$RequestExecutor.doExecute(AmazonHttpClient.java:758)
        at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeWithTimer(AmazonHttpClient.java:732)
        at com.amazonaws.http.AmazonHttpClient$RequestExecutor.execute(AmazonHttpClient.java:714)
        at com.amazonaws.http.AmazonHttpClient$RequestExecutor.access$500(AmazonHttpClient.java:674)
        at com.amazonaws.http.AmazonHttpClient$RequestExecutionBuilderImpl.execute(AmazonHttpClient.java:656)
        at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:520)
        at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:4705)
        at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:4652)
        at com.amazonaws.services.s3.AmazonS3Client.doUploadPart(AmazonS3Client.java:3657)
        at com.amazonaws.services.s3.AmazonS3Client.uploadPart(AmazonS3Client.java:3642)
        at com.amazonaws.services.s3.transfer.internal.UploadCallable.uploadPartsInSeries(UploadCallable.java:258)
        at com.amazonaws.services.s3.transfer.internal.UploadCallable.uploadInParts(UploadCallable.java:189)
        at com.amazonaws.services.s3.transfer.internal.UploadCallable.call(UploadCallable.java:121)
        at com.amazonaws.services.s3.transfer.internal.UploadMonitor.call(UploadMonitor.java:143)
        at com.amazonaws.services.s3.transfer.internal.UploadMonitor.call(UploadMonitor.java:48)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
```

My code is similar to @mstockm an is simply uploading a file (currently testing with a 20GB file) from HDFS into S3 with SSE-AES256: 

```java
TransferManager tm = TransferManagerBuilder.standard().withS3Client(s3Client).build();
ObjectMetadata objMeta = new ObjectMetadata();
objMeta.setSSEAlgorithm(ObjectMetadata.AES_256_SERVER_SIDE_ENCRYPTION);
long fileLength = fs.getFileStatus(pathToFileInHDFS).getLen();
objMeta.setContentLength(fileLength);
PutObjectRequest putRequest = new PutObjectRequest(destinationBucket, destinationKey, inputStream, objectMetadata);
Upload upload = tm.upload(putRequest);
upload.waitForCompletion(); 
```

Let me know if I should make a separate issues for this instead! Thank you all so much!

**Comment by debora-ito** [comment]
@cl305 I'm taking a look.

**Comment by cl305** [comment]
@debora-ito Hey, so it turns out that regardless of which version of the AWS SDK we set it to, it always seems to be 1.11.479 running. This is because we're executing our jobs on an EMR built with version 5.21.0, which has a default AWS SDK version of 1.11.479 (this is probably the first class path that our jar finds when it executes, instead of whatever version we set it to in our pom.xml). I'm going to create a new EMR with 5.22.0 and hopefully that should solve the issues that I'm seeing. Thanks a lot!
